app/services/account_search_service.rb

```diff
@@ -151,13 +151,23 @@ def core_query
   end
 
   def call(query, account = nil, options = {})
-    @query   = query&.strip&.gsub(/\A@/, '')
-    @limit   = options[:limit].to_i
-    @offset  = options[:offset].to_i
-    @options = options
-    @account = account
+    MastodonOTELTracer.in_span('AccountSearchService#call') do |span|
+      @query   = query&.strip&.gsub(/\A@/, '')
+      @limit   = options[:limit].to_i
+      @offset  = options[:offset].to_i
+      @options = options
+      @account = account
+
+      span.add_attributes(
+        'search.offset' => @offset,
+        'search.limit' => @limit,
+        'search.backend' => Chewy.enabled? ? 'elasticsearch' : 'database'
+      )
 
-    search_service_results.compact.uniq
+      results = search_service_results.compact.uniq
```

Comment:
Minor style nitpick here, you can use `#tap` to avoid needing the local var assigngment and explicit return, something like:

```ruby
search_service_results.compact.uniq.tap do |results|
  span.set_attribute('search.results.count', results.size)
end
```

Reply:
I'm a fan of `.tap` for moments like these, too. What do you think of doing all the addition once in that last `.tap`?

```ruby
  def call(query, account = nil, options = {})
    MastodonOTELTracer.in_span('AccountSearchService#call') do |span|
      @query   = query&.strip&.gsub(/\A@/, '')
      @limit   = options[:limit].to_i
      @offset  = options[:offset].to_i
      @options = options
      @account = account

      search_service_results.compact.uniq.tap do |results|
        span.add_attributes({
          'search.offset' => @offset,
          'search.limit' => @limit,
          'search.backend' => Chewy.enabled? ? 'elasticsearch' : 'database',
          'search.results.count' => r.size
        })
      end
    end
```

Reply:
I split the attributes as I thought it would make sense to have some attributes set before the main body of work is done, so they are set on the span even if the span throws an exception, which will allow to inspect the error-ed span and the attributes might be useful for troubleshooting.

Reply:
> [known attributes] on the span even if the [main body of work] throws an exception

… That is a fantastic point. Yes. Do that.

… I wrote a variation here with the early attributes assigned as the `attributes:` keyword argument to `<tracer>.in_span()` and deleted it because it got ugly.

What you've got works! And @mjankowski's tap at the end would have the benefits he described: avoid a local variable† and explicit return.

† And memory allocation for the local variable? 🤔